sw/device/tests/autogen/plic_all_irqs_test.c

```diff
@@ -0,0 +1,183 @@
+
+  // Clear the interrupt at UART.
+  CHECK(dif_uart_irq_acknowledge(uart, uart_irq) == kDifUartOk,
+        "dif_uart_irq_acknowledge failed");
+}
+
+/**
+ * Provides external irq handling for this test.
+ *
+ * This function overrides the default external irq handler in
+ * `sw/device/lib/handler.h`.
+ */
+void handler_irq_external(void) {
+  // Find which interrupt fired at PLIC by claiming it.
+  dif_rv_plic_irq_id_t plic_irq_id;
+  CHECK(dif_rv_plic_irq_claim(&plic, kHart, &plic_irq_id) == kDifRvPlicOk,
+        "dif_rv_plic_irq_claim failed");
+
+  // Check if it is the right peripheral.
+  top_earlgrey_plic_peripheral_t peripheral = (top_earlgrey_plic_peripheral_t)
+      top_earlgrey_plic_interrupt_for_peripheral[plic_irq_id];
+  CHECK(peripheral == peripheral_expected,
+        "Interurpt from incorrect peripheral: exp = %d, obs = %d",
+        peripheral_expected, peripheral);
+
+  switch (peripheral) {
+    case kTopEarlgreyPlicPeripheralUart0:
+      uart_isr(&uart0, plic_irq_id);
+      break;
+    default:
+      LOG_FATAL("ISR is not implemented!");
+      test_status_set(kTestStatusFailed);
+  }
+
+  // Complete the irq at PLIC.
+  CHECK(dif_rv_plic_irq_complete(&plic, kHart, plic_irq_id) == kDifRvPlicOk,
+        "dif_rv_plic_irq_complete failed");
+}
+
+/**
+ * Enables all irqs at PLIC.
+ */
+static void plic_irqs_enable(void) {
+  mmio_region_t plic_base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR);
+  CHECK(dif_rv_plic_init((dif_rv_plic_params_t){.base_addr = plic_base_addr},
+                         &plic) == kDifRvPlicOk,
+        "dif_rv_plic_init failed");
+
+  for (top_earlgrey_plic_irq_id_t irq_id = kTopEarlgreyPlicIrqIdNone + 1;
+       irq_id <= kTopEarlgreyPlicIrqIdLast; ++irq_id) {
+    // TODO: pick priority randomly between 0-3.
+    CHECK(dif_rv_plic_irq_set_priority(&plic, irq_id, 0x1) == kDifRvPlicOk,
+          "dif_rv_plic_irq_set_priority failed");
+
+    CHECK(dif_rv_plic_irq_set_enabled(&plic, irq_id, kHart,
+                                      kDifRvPlicToggleEnabled) == kDifRvPlicOk,
+          "dif_rv_plic_irq_set_enabled failed");
+  }
+  CHECK(dif_rv_plic_target_set_threshold(&plic, kHart, 0x0) == kDifRvPlicOk,
+        "dif_rv_plic_target_set_threshold failed");
+}
+
+/**
+ * Enables all irqs in all peripherals.
+ */
+static void peripheral_irqs_enable(void) {
+  // UART0.
+  mmio_region_t uart_base_addr =
+      mmio_region_from_addr(TOP_EARLGREY_UART0_BASE_ADDR);
+  CHECK(dif_uart_init(
+            (dif_uart_params_t){
+                .base_addr = uart_base_addr,
+            },
+            &uart0) == kDifUartOk,
+        "dif_uart_init failed");
+  for (dif_uart_irq_t uart_irq = kDifUartIrqTxWatermark;
+       uart_irq <= kDifUartIrqRxParityErr; ++uart_irq) {
+    CHECK(dif_uart_irq_set_enabled(&uart0, uart_irq, kDifUartToggleEnabled) ==
+              kDifUartOk,
+          "dif_uart_irq_set_enabled failed");
+  }
```

Comment:
@timothytrippel, it will be nice to have one function that just enables all interrupts as a DIF.

Reply:
As you suggested, just use dif_*_irq_restore_all(<all-1s>).